⚠️ CONFLICT! Lineage pull request for: skeleton

felddy opened the pull request:
# Lineage Pull Request: CONFLICT #

![DANGER](https://raw.githubusercontent.com/cisagov/action-lineage/develop/src/achtung.gif)

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-docker.git`](https://github.com/cisagov/skeleton-docker.git)


Check the changes in this pull request to ensure they won't cause issues with
your project.

The `lineage/skeleton` branch has **one or more unresolved merge conflicts**
that you must resolve before merging this pull request!

## How to resolve the conflicts ##

1. Clone the `lineage/skeleton` branch locally:

    ```console
    git clone --single-branch --branch lineage/skeleton git@github.com:cisagov/pca-gophish-composition.git
    ```

1. Review each of the following files looking for [conflict-resolution markers](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging#_basic_merge_conflicts):

    ```console
    .bandit.yml
    .github/workflows/build.yml
    .gitignore
    .isort.cfg
    .pre-commit-config.yaml
    CONTRIBUTING.md
    README.md
    docker-compose.yml
    requirements-test.txt
    tests/conftest.py
    tests/container_test.py
    ```

1. Resolve each of the conflicts and `add` your changes to the branch:

    ```console
    git add .bandit.yml .github/workflows/build.yml .gitignore .isort.cfg .pre-commit-config.yaml CONTRIBUTING.md README.md docker-compose.yml requirements-test.txt tests/conftest.py tests/container_test.py 
    ```

1. Once all the changes have been added, `commit` and `push` the changes back to
GitHub:

    ```console
    git commit --message="Resolved Lineage conflicts."
    git push
    ```

1. Wait for all the automated tests to pass.

1. Check the "Everything is cool" checkbox below:

    - [x] ✌️ The conflicts in this pull request have been resolved.

1. Mark this draft pull request "Ready for review".

------------

**Note:** *You are seeing this because one of this repository's maintainers has
configured [Lineage] to open pull requests.*

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

felddy committed:
Split bandit pre-commit checks into two diffrent runs

One for tests and one for everthing else.

felddy committed:
Merge pull request #7 from cisagov/improvement-pytest

Improvement pytest

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge pull request #9 from cisagov/improvement/upstream_updates

Upstream updates

felddy committed:
update pre-commit hooks

felddy committed:
Add initial postfix templates and configurations

felddy committed:
Add password secret for mailarchive user

felddy committed:
Sort postfix config and add proper environment variables

felddy committed:
Add configurations for supporting services

felddy committed:
Update docker components

felddy committed:
Add custom docker entrypoint

felddy committed:
Move templates and secrets into src

felddy committed:
Update URLs

felddy committed:
Get ready for docker hub integration

felddy committed:
Add scripts to support travis

felddy committed:
Add container info to README

felddy committed:
Remove uneeded

felddy committed:
Add a version file to bump

felddy committed:
Add magic keys for travis deploy

felddy committed:
Remove install section

felddy committed:
Re-enable pre-commit tests

felddy committed:
Update README

felddy committed:
Add real image name to docker-compose build

felddy committed:
Move container build to install instead of script.
This allows a hard fail of the container doesn't build.

felddy committed:
Limit bandit in tests tree

felddy committed:
Add container pytests

felddy committed:
add ignore for pytest caches

felddy committed:
add latest tag to build for use by compose

felddy committed:
Remove check that was failing due to race

felddy committed:
Add diceware for password generator

felddy committed:
Change secrets to new user.txt file

felddy committed:
Add dynamic user generator and tests.

felddy committed:
Update documentation for user secrets

felddy committed:
Bump version

felddy committed:
Add version slices to docker deploy

felddy committed:
Spell out flag names

felddy committed:
Rename repo

felddy committed:
Spell out flag names

felddy committed:
Remove non-skel files

felddy committed:
revert version

felddy committed:
Update project name

felddy committed:
Add data to the package.
This shows how to added data to a package and correctly access it.

felddy committed:
Add reading of package data and environment

felddy committed:
Rename project and stow ciphertext

felddy committed:
Add example docker composition for testing

felddy committed:
Add sample tests for container using pytest-dockerc

felddy committed:
Add a secret quote

felddy committed:
Add dockerfile to build example container

felddy committed:
Update README

felddy committed:
Remove composition up and down.
This is now handled by pytest itself

felddy committed:
Update ciphertext for DOCKER_PW

felddy committed:
Remove branch checkout

felddy committed:
Declare setuptools explicity as a requirement for pkg_resources use.

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_in_upstream_changes

jsf9k committed:
Merge pull request #11 from cisagov/improvement/merge_in_upstream_changes

Merge in upstream changes from cisagov/skeleton-generic

felddy committed:
Merge branch 'develop' into improvement/data

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_in_upstream_changes

jsf9k committed:
Merge pull request #1 from cisagov/improvement/merge_in_upstream_changes

Merge in upstream changes from cisagov/skeleton-generic

felddy committed:
Merge pull request #10 from cisagov/improvement/data

Improvement/data

felddy committed:
Add version label

felddy committed:
Add robust container tests

felddy committed:
Add second instance of container for version test

felddy committed:
Add useful labels

felddy committed:
Use release of code instead of repo

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/pull_in_upstream_changes

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/pull_in_upstream_changes

felddy committed:
Expose ARGs for external configuration

felddy committed:
Update to show version specification for manual build

felddy committed:
Add build arguments for image creation

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-docker into develop

felddy committed:
Fix for travis being a bit pokey at startup

felddy committed:
Add version check for TRAVIS_TAG when available

felddy committed:
Travis you biatch!

felddy committed:
Travis you biatch!!

felddy committed:
You're not wrong Travis, you're just an 🐴🕳.

jsf9k committed:
Merge pull request #12 from cisagov/improvement/pull_in_upstream_changes

Pull in upstream changes

jsf9k committed:
Merge pull request #2 from cisagov/improvement/pull_in_upstream_changes

Pull in upstream changes

felddy committed:
Add missing run instructions

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-docker into develop

jsf9k committed:
Add Snyk badge/shield

jsf9k committed:
Merge pull request #13 from cisagov/improvement/add_snyk_badge

Add Snyk badge/shield

felddy committed:
Add test to verify TRAVIS_TAG agrees with project version during release

felddy committed:
Merge branch 'develop' into improvement/travis_version_test

felddy committed:
Separate release version test from stdout version test

felddy committed:
TRAVIS_TAG must start with a "v"

felddy committed:
Merge pull request #14 from cisagov/improvement/travis_version_test

Add test to verify TRAVIS_TAG agrees with project version during release

felddy committed:
Add lgtm configuration for requirements-test.txt file

felddy committed:
Add LGTM shields

felddy committed:
Merge pull request #3 from cisagov/improvement/lgtm

Improvement/lgtm

mcdonnnj committed:
Pull and merge updates from skeleton-generic.

mcdonnnj committed:
Merge pull request #4 from cisagov/add_skeleton-generic_updates

Add skeleton generic updates

mcdonnnj committed:
Merging updates to skeleton-generic.

mcdonnnj committed:
Merge pull request #15 from cisagov/sync_skeleton-generic_updates

Sync skeleton generic updates

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_changes_from_skeleton_generic

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_changes_from_skeleton_generic

jsf9k committed:
Merge pull request #6 from cisagov/improvement/upstream_changes_from_skeleton_generic

Upstream changes from cisagov/skeleton-generic

jsf9k committed:
Merge pull request #16 from cisagov/improvement/upstream_changes_from_skeleton_generic

Upstream changes from cisagov/skeleton-generic

felddy committed:
Add test to check for container readiness.

felddy committed:
Merge pull request #7 from cisagov/improvement/wait_for_ready_test

Add test to check for container readiness.

felddy committed:
Add script to push readme to Docker Hub.
Snarfed from: 
https://github.com/moikot/golang-dep/blob/aab3ea8462a19407544f1ce9daa11c3f0924394c/.travis/push.sh#L127

felddy committed:
Add comment to identify ciphertext blob purpose

felddy committed:
Add call to push_readme script

felddy committed:
Document script inputs

felddy committed:
Merge pull request #8 from cisagov/improvement/publish_readme_to_docker_hub

Improvement/publish readme to docker hub

felddy committed:
Register mark in heed of pytest warning.
https://docs.pytest.org/en/latest/mark.html

felddy committed:
Merge pull request #17 from cisagov/fix/pytest_warning

Register mark in heed of pytest warning.

felddy committed:
fix issue 5

felddy committed:
Merge pull request #9 from cisagov/fix/test_split

Spit compound version test into separate tests

felddy committed:
Open the README file with utf-8 encoding.
Backported from pca-gophish-composition.

felddy committed:
Merge pull request #18 from cisagov/improvement/utf-8-readme

Open the README file with utf-8 encoding.

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge pull request #10 from cisagov/improvement/upstream_updates

Upstream updates

jsf9k committed:
Merge pull request #19 from cisagov/improvement/upstream_updates

Upstream updates

felddy committed:
Add a actions workflow similar to the travis config.

felddy committed:
Change python version to 3.8

felddy committed:
Revert to python 3.7

felddy committed:
Work around pre-commit issue. 
See: https://github.com/pypa/setuptools/issues/1869

felddy committed:
Fix install_requires syntax.

felddy committed:
Update pre-commit config to lastest versions.

felddy committed:
Remove setuptools pinning.
This is an ansible lint issue: 
https://github.com/ansible/ansible-lint/issues/590

felddy committed:
Apply workaround for ansible-lint error.
See: https://github.com/ansible/ansible-lint/issues/590

felddy committed:
Add special sauce for coveralls.
See: https://github.com/marketplace/actions/coveralls-github-action

felddy committed:
Add config to tell coveralls where the coverage file lives.

felddy committed:
Fix syntax error.  Learning is fun!

felddy committed:
Switch to using coveralls with token.

felddy committed:
Rename workflow and add badge to README.

felddy committed:
Correct capitalization for badge.

felddy committed:
Make action badge a link.

felddy committed:
Lowercasing workflow name so the shield matches all others.

felddy committed:
Remove travis. :(

felddy committed:
Add comment about why we are pinning virtualenv

felddy committed:
Rename env variable to make sense in a post-travis world.

felddy committed:
Formatting

felddy committed:
Merge pull request #20 from cisagov/improvement/actions

Switch CI from Travis to GitHub actions

felddy committed:
Split workflow into multiple jobs.  Add artifact output.

felddy committed:
Merge pull request #22 from cisagov/improvement/split_actions

Split workflow into multiple jobs.  Add artifact output.

mcdonnnj committed:
Pull upstream changes from skeleton-generic.

mcdonnnj committed:
Remove virtualenv version pinning as the issue has been resolved in ansible-lint and the updated version is included in the version being used now.

mcdonnnj committed:
Merge pull request #23 from cisagov/pull_upstream_changes_2019-11-05

Pull upstream changes from skeleton-generic.

felddy committed:
Add caches for GitHub Actions.

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into develop

felddy committed:
Add new workflow: lint, test, and build artifact.

felddy committed:
Pass in the github release tag for testing.

felddy committed:
Merge pull request #25 from cisagov/improvement/action_cache

Add caches for GitHub Actions.

felddy committed:
Add release workflow based on Travis.

felddy committed:
Remove Travis configuration.

felddy committed:
Create a version lookup step with output.

felddy committed:
Merge branch 'develop' into bug/release_tag

felddy committed:
Reorder jobs and pass image artifacts.

felddy committed:
Simplify artifact naming.

felddy committed:
Change environment variable name to be CI agnostic.

felddy committed:
Move images from dhsncats to cisagov docker hub org.

felddy committed:
Merge pull request #12 from cisagov/improvement/actions

Migrate CI system and Docker Hub organization

felddy committed:
Pass in the github release tag for testing.

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-docker into improvement/actions

felddy committed:
Merge pull request #13 from cisagov/improvement/actions

Add missing env variable for test

felddy committed:
Applying autoupdate of pre-commit hooks.
Just trying to tickle GitHub Actions so my tests will restart.

felddy committed:
Merge branch 'bug/release_tag' of github.com:cisagov/skeleton-python-library into bug/release_tag

felddy committed:
Merge pull request #26 from cisagov/bug/release_tag

Fix missing release tag being passed into tests.

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/upstream

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/upstream

felddy committed:
Merge pull request #14 from cisagov/improvement/upstream

Improvement/upstream

felddy committed:
Merge pull request #27 from cisagov/improvement/upstream

Improvement/upstream

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/codeowners

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/codeowners

felddy committed:
Merge pull request #28 from cisagov/improvement/codeowners

Improvement/codeowners

felddy committed:
Merge pull request #15 from cisagov/improvement/codeowners

Improvement/codeowners

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_and_run_isort

mcdonnnj committed:
Ran pre-commit to populate .isort.cfg and let isort run against the repository.

mcdonnnj committed:
Add "coverage < 5.0" to setup.py to enforce it because coveralls does not currently support coverage 5.0+ per https://github.com/coveralls-clients/coveralls-python/issues/203

mcdonnnj committed:
Merge pull request #30 from cisagov/improvements/pull_in_upstream_and_run_isort

Pull in Upstream and Run pre-commit

mcdonnnj committed:
Add bump_version.sh and tag.sh to support bumping versions and tagging through scripts. Update setup.py to include the semver package which is used by the bump_version.sh script.

mcdonnnj committed:
Move semver requirement to test from install.

mcdonnnj committed:
Move semver requirement from test requirements in setup.py to requirements-dev.txt file to better match intent.

mcdonnnj committed:
Add comments explaining why a temp file is used when updating the version in the bump_version.sh script.

mcdonnnj committed:
Merge pull request #31 from cisagov/improvement/add_version_bump_and_tag_scripts

Add Scripts for Bumping Versions and Tagging

mcdonnnj committed:
Add python_requires corresponding to listed Python versions in setup.py
Add versioning for setuptools to support inclusiong of python_requires argument.
Note: This will require pip versions >=9.0.0 for support.

mcdonnnj committed:
Merge pull request #32 from cisagov/improvement/add_python_requires

Add python_requires argument for setup.py

mcdonnnj committed:
Pull in changes to build workflow from skeleton-generic to support correct workflow if someone forks and does a pull request.

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream-changes

jsf9k committed:
Add changes made by isort pre-commit hook

jsf9k committed:
Merge pull request #16 from cisagov/improvement/upstream-changes

Pull in upstream changes

mcdonnnj committed:
Merge pull request #33 from cisagov/improvement/pull_build_workflow_change_from_upstream

Pull in build Workflow Changes from Upstream

felddy committed:
Add schema library.

felddy committed:
Add schema checks for arguments.
Log output instead of print.
Improve docs.
Do some interesting parsing.

felddy committed:
Add tests for new arguments

felddy committed:
Removed uses of xfail.
This was a bad example of when to use xfail.  See the semantics here:
http://doc.pytest.org/en/latest/skipping.html

felddy committed:
Bumping github actions python version to 3.8.
Advertise support for 3.8 in setup.py

felddy committed:
Merge pull request #34 from cisagov/improvement/schema

Add schema validation for docopt arguments.

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_updates

mcdonnnj committed:
Fix spacing before comment in .bandit.yml (shown with pre-commit run --verbose --all-files).

mcdonnnj committed:
Unpin coverage as coveralls supports coverage 5.0+ as of 1.10.0

mcdonnnj committed:
Add --verbose flag to the coveralls call in the test workflow.

mcdonnnj committed:
Pin coveralls version to see if the service_number addition in 1.11.0 is breaking coveralls uploads.

mcdonnnj committed:
Remove --verbose from coveralls call in test workflow.
Add comment explaining why coveralls is being pinned.

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_updates

mcdonnnj committed:
Add missing space in .bandit.yml

mcdonnnj committed:
Pin to testing branch per https://github.com/coveralls-clients/coveralls-python/issues/219 for testing.

mcdonnnj committed:
Change coveralls pin to avoid the version with the regression. 1.11.1 is released which fixed the regression introduced in 1.11.0 per https://github.com/coveralls-clients/coveralls-python/releases/tag/1.11.1

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_updates

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_updates

mcdonnnj committed:
Adjust language in description of pre-commit step in the GitHub Actions workflow.

mcdonnnj committed:
Merge pull request #17 from cisagov/improvements/pull_in_upstream_updates

Pull in updates from skeleton-generic

mcdonnnj committed:
Merge pull request #36 from cisagov/improvements/pull_in_upstream_updates

Pull in Updates from skeleton-generic

felddy committed:
Autoupdate pre-commit hooks.  Add mypy.

felddy committed:
Sort .gitignore add mypy cache.

felddy committed:
Add PEP 484 type hints.

See: https://www.python.org/dev/peps/pep-0484/

felddy committed:
Merge pull request #37 from cisagov/improvement/static_type_checking

Add static type checking to the Python skeleton

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into bugfix/pull_in_pre-commit_fix

mcdonnnj committed:
Merge pull request #38 from cisagov/bugfix/pull_in_pre-commit_fix

Pull in Fix for pre-commit Caching from Upstream

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_upstream_changes

mcdonnnj committed:
Merge pull request #18 from cisagov/improvements/pull_upstream_changes

Pull in Upstream Changes

mcdonnnj committed:
Fixed broken hook id for pre-commit-terraform:
  terraform_validate_no_variables was changed to terraform_validate in the
  following commit:
  https://github.com/antonbabenko/pre-commit-terraform/commit/35e0356188b64a4c5af9a4e7200d936e514cba71.
Ran pre-commit autoupdate.

mcdonnnj committed:
Merge pull request #40 from cisagov/improvements/update_pre-commit

Update pre-commit Hooks and Fix Broken Hook ID

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/upstream_updates

mcdonnnj committed:
Merge pull request #19 from cisagov/improvements/upstream_updates

Pull in Upstream Updates

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/upstream_updates

mcdonnnj committed:
Merge pull request #39 from cisagov/improvements/upstream_updates

Pull in Upstream Updates

mcdonnnj committed:
Ran `pre-commit autoupdate` to get the latest version of the
`pre-commit-terraform` hook since the PR I submitted,
https://github.com/antonbabenko/pre-commit-terraform/pull/100,
was approved. This will fix issues with `skeleton-tf-module` related to multiple
directories with Terraform code.

mcdonnnj committed:
Explicitly install pre-commit hooks as its own step so it's clearer when the
failure is with setup rather than hooks running.

mcdonnnj committed:
Rename action to accurately reflect that we are only linting, not building.
Adjust description for pre-commit hook setup to better convey what is being
done.

mcdonnnj committed:
Sort .gitignore entries.

mcdonnnj committed:
Merge pull request #41 from cisagov/bugfix/update_pre-commit_for_pre-commit-terraform_hook

Update pre-commit Hooks to Include Bugfix

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_upstream_updates

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_upstream_updates

mcdonnnj committed:
Remove duplicate '.mypy_cache' entry in .gitignore

mcdonnnj committed:
Merge pull request #20 from cisagov/improvements/pull_upstream_updates

Pull in Upstream Updates

mcdonnnj committed:
Merge pull request #40 from cisagov/improvements/pull_upstream_updates

Pull in Upstream Updates

hillaryj committed:
Update CODEOWNERS

hillaryj committed:
Merge pull request #42 from cisagov/hillaryj-owner

Update CODEOWNERS

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream-changes

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream-changes

jsf9k committed:
Merge pull request #41 from cisagov/improvement/upstream-changes

Pull in upstream changes from cisagov/skeleton-generic

jsf9k committed:
Merge pull request #21 from cisagov/improvement/upstream-changes

Pull in upstream changes from cisagov/skeleton-generic

felddy committed:
Allow events from apb to rebuild this repository weekly.

felddy committed:
Merge pull request #43 from cisagov/improvement/apb-events

Allow events from apb to rebuild this repository weekly.

felddy committed:
Add Lineage configuration.

felddy committed:
Add Lineage configuration.

felddy committed:
Merge pull request #42 from cisagov/improvement/lineage

Add Lineage configuration.

felddy committed:
Merge pull request #22 from cisagov/improvement/lineage

Add Lineage configuration.

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

felddy committed:
Merge pull request #23 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

Commit:
Merge https://github.com/cisagov/skeleton-docker into lineage/skeleton

# Conflicts:
#	.bandit.yml
#	.github/lineage.yml
#	.github/workflows/build.yml
#	.gitignore
#	.isort.cfg
#	.pre-commit-config.yaml
#	CONTRIBUTING.md
#	README.md
#	docker-compose.yml
#	requirements-test.txt
#	tests/conftest.py
#	tests/container_test.py

felddy commented:
Jumping Jesus on a pogo stick!

felddy committed:
Resolve lineage PR conflicts.

felddy committed:
Remove mismerged test.

jsf9k commented on Dockerfile:
```diff
@@ -0,0 +1,42 @@
+
+WORKDIR ${CISA_HOME}
+
+RUN wget -O sourcecode.tgz https://github.com/cisagov/skeleton-python-library/archive/v${VERSION}.tar.gz && \
```
This looks incorrect.  Is this file needed?

felddy replied:
Egads!  You are correct!  I missed the entire clean file additions that came in.

jsf9k commented on bump_version.sh:
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
```
If this file exists then don't we need to pull in `semver`, perhaps via `requirements-dev.txt`?  Usually we put `ipython` in `requirements-dev.txt` as well.  I'm not sure why that file got removed from this PR.

jsf9k reviewed (changes requested):
I think there are at least a few errors/omissions here.

felddy commented on .github/workflows/build.yml:
```diff
@@ -1,12 +1,14 @@
+    types: [apb]
 
 env:
+  IMAGE_NAME: cisagov/example
```
This is wrong.

felddy commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,53 @@
+    types: [prereleased, released]
+
+env:
+  IMAGE_NAME: cisagov/example
```
WRONG!

felddy replied:
![SlsSmSr](https://user-images.githubusercontent.com/3229435/81943082-eff48a00-95c8-11ea-8e6d-c0711f911335.png)

felddy replied:
I was going to change this back to "Draft" status, but the button isn't working.  Maybe because of the "changes requested" state.  In any case, I apologize for this crap ass merge job I did.  I'll clean this up once I get off dad-doody.

jsf9k replied:
No worries.  This very thang caught me slippin' a few days ago in another repo when I switched from `skeleton-packer-cool` to `skeleton-packer` as the skeleton.

Doing `git status` and `git diff` right after the `git merge` and checking that all the cleanly merged files make sense is key.  `git` handles merges like a champ, but it doesn't always do what you want.  It is good to be reminded of these things from time to time.

felddy committed:
Repair previous merge errors.

The crux of the problem was that the lineage was not truly 
skeleton-docker.  This is not a Docker project.  It probably should be a 
skeleton-python-library ancestor.  Also that I wasn't paying close 
attention.

felddy committed:
Change lineage to correct parent. Pull in changes from sk-py-lib

felddy committed:
Make git ignores agree with upstream.

felddy commented:
Ok.  Let's try this again.  A lot of the pain was caused by a mismatch in the lineage.  This project is not a Docker container.  It is more akin to a Python library.  Once I made that change the pieces fell into place better. 

Again... I apologize for the previous state of the merge.

jsf9k reviewed (approved):
I _think_ this merge is OK now.

dav3r commented on src/version.txt:
```diff
@@ -0,0 +1 @@
+__version__ = "0.0.1"
```
Shouldn't we use `src/gophish_init/_version.py` as the version file for this repo?

dav3r commented on bump_version.sh:
```diff
@@ -0,0 +1,51 @@
+set -o errexit
+set -o pipefail
+
+VERSION_FILE=src/version.txt
```
Shouldn't we use `src/gophish_init/_version.py` as the version file for this repo?

dav3r reviewed (approved):
Everything looks good, except for the version file stuff I mentioned below.
👍

felddy committed:
Point bump_version to correct version locations.  Remove unused file.
